src/panels/AzureServiceOperatorPanel.ts

```diff
@@ -0,0 +1,264 @@
+
+export class AzureServiceOperatorPanel extends BasePanel<"aso"> {
+    constructor(extensionUri: vscode.Uri) {
+        super(extensionUri, "aso");
+    }
+}
+
+export class AzureServiceOperatorDataProvider implements PanelDataProvider<"aso"> {
+    constructor(
+        readonly extension: vscode.Extension<vscode.ExtensionContext>,
+        readonly kubectl: k8s.APIAvailable<k8s.KubectlV1>,
+        readonly kubeConfigFilePath: string,
+        readonly azAccount: AzureAccountExtensionApi,
+        readonly clusterName: string
+    ) { }
+
+    getTitle(): string {
+        return `ASO on ${this.clusterName}`;
+    }
+
+    getInitialState(): InitialState {
+        return {
+            clusterName: this.clusterName
+        };
+    }
+
+    getMessageHandler(webview: MessageSink<ToWebViewMsgDef>): MessageHandler<ToVsCodeMsgDef> {
+        return {
+            checkSPRequest: args => this._handleCheckSPRequest(args.appId, args.appSecret, webview),
+            installCertManagerRequest: _ => this._handleInstallCertManagerRequest(webview),
+            waitForCertManagerRequest: _ => this._handleWaitForCertManagerRequest(webview),
+            installOperatorRequest: _ => this._handleInstallOperatorRequest(webview),
+            installOperatorSettingsRequest: args => this._handleInstallOperatorSettingsRequest(args.appId, args.appSecret, args.cloudName, args.subscriptionId, args.tenantId, webview),
+            waitForControllerManagerRequest: _ => this._handleWaitForControllerManagerRequest(webview)
+        };
+    }
+
+    private async _handleCheckSPRequest(appId: string, appSecret: string, webview: MessageSink<ToWebViewMsgDef>): Promise<void> {
+        const servicePrincipalAccess = await getServicePrincipalAccess(this.azAccount, appId, appSecret);
+        if (failed(servicePrincipalAccess)) {
+            webview.postMessage({
+                command: "checkSPResponse",
+                parameters: {
+                    succeeded: false,
+                    errorMessage: servicePrincipalAccess.error,
+                    commandResults: [],
+                    cloudName: null,
+                    subscriptions: [],
+                    tenantId: null
+                }
+            });
+            return;
+        }
+
+        webview.postMessage({
+            command: "checkSPResponse",
+            parameters: {
+                succeeded: true,
+                errorMessage: null,
+                commandResults: [],
+                cloudName: servicePrincipalAccess.result.cloudName as AzureCloudName,
+                subscriptions: servicePrincipalAccess.result.subscriptions,
+                tenantId: servicePrincipalAccess.result.tenantId
+            }
+        });
+    }
+
+    private async _handleInstallCertManagerRequest(webview: MessageSink<ToWebViewMsgDef>): Promise<void> {
+        const asoCrdYamlFile = "https://github.com/jetstack/cert-manager/releases/download/v1.12.1/cert-manager.yaml";
```

Comment:
💡 Observation ☕️☕️🥷 - I wonder what is the difference and if raw version of this file is there via `raw.gh url`  https://github.com/cert-manager/cert-manager/releases/download/v1.12.1/cert-manager.yaml

Reply:
This URL is taken from the [ASO installation instructions](https://azure.github.io/azure-service-operator/#installation).

I _think_ the `raw.githubusercontent` URLs are for files which exist within the repo. The `yaml` file referred to here is a published asset (so it's built from the repo but may not be present in there).

Interesting that there is a redirect from `https://github.com/jetstack/cert-manager` to `https://github.com/cert-manager/cert-manager`, though. Maybe the ASO docs should be updated, but for now I think we can stick as closely as possible with the published docs; we can easily update this later if necessary.

Reply:
Sounds good to me🍿 might be add the official docs link here and then resolve conversation for archival sake.

Reply:
Added